bundles/org.openhab.binding.serial/src/main/java/org/openhab/binding/serial/internal/handler/SerialDeviceHandler.java

```diff
@@ -0,0 +1,179 @@
+                if (bridge != null) {
+                    final SerialBridgeHandler handler = (SerialBridgeHandler) bridge.getHandler();
+                    if (handler != null) {
+                        channel.mapCommand(command).ifPresent(value -> handler.writeString(value));
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        final SerialDeviceConfiguration config = getConfigAs(SerialDeviceConfiguration.class);
+
+        try {
+            devicePattern = Pattern.compile(config.patternMatch);
+        } catch (final PatternSyntaxException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Invalid device pattern: " + e.getMessage());
+            return;
+        }
+
+        for (final Channel c : getThing().getChannels()) {
+            final ChannelTypeUID type = c.getChannelTypeUID();
+            if (type != null) {
+                final ChannelConfig channelConfig = c.getConfiguration().as(ChannelConfig.class);
+                try {
+                    final DeviceChannel deviceChannel = DeviceChannelFactory
+                            .createDeviceChannel(valueTransformationProvider, channelConfig, type.getId());
+                    if (deviceChannel != null) {
+                        channels.put(c.getUID(), deviceChannel);
+                    }
+                } catch (final IllegalArgumentException e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Configuration error for channel " + c.getUID().getId() + ": " + e.getMessage());
+                    return;
+                }
+            }
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+        bridgeStatusChanged(getBridgeStatus());
+    }
+
+    @Override
+    public void dispose() {
+        channels.clear();
+        lastValue = null;
+        super.dispose();
+    }
+
+    /**
+     * Handle a line of data received from the bridge
+     *
+     * @param data the line of data
+     */
+    public void handleData(final String data) {
+        final Pattern devicePattern = this.devicePattern;
+
+        if (devicePattern != null && devicePattern.matcher(data).matches()) {
+            channels.forEach((channelUID, channel) -> refresh(channelUID, channel, data));
+            this.lastValue = data;
+        }
+    }
+
+    @Override
+    public void bridgeStatusChanged(final ThingStatusInfo bridgeStatusInfo) {
+        if (getThing().getStatusInfo().getStatusDetail() == ThingStatusDetail.CONFIGURATION_ERROR) {
+            return;
+        }
+
+        if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE && getThing().getStatus() == ThingStatus.UNKNOWN) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
+            return;
+        }
+
+        super.bridgeStatusChanged(bridgeStatusInfo);
+    }
```

Comment:
The `BaseThingHandler` sets the Thing states accordingly when the bridge status changes. Is this method necessary?

Reply:
This is preventing the thing from going online if there is a thing level configuration error.

Reply:
I understood that the Thing should be online when there's no configuration error and the Bridge is online.

This behavior can be achieved by using the framework. You can set the Thing status to ONLINE when there is no configuration error. Then, the framework takes care, that the Thing is only online when the Bridge is.

Reply:
It seems the framework only takes care of it if you explicitly check the bridge status and don't put the thing online if the bridge is offline. The framework allows you to set the thing online even though the bridge is offline. I'll make that change since its cleaner not overriding `bridgeStatusChanged`.

Reply:
From my knowledge the Thing doesn't go online when the bridge is offline, although if the binding sets it explicitly to online. But I'm open for conviction.